hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/publishing/handlers/TopicHandler.java

```diff
@@ -55,10 +56,14 @@ private void onTopicPresent(HttpServerExchange exchange, String messageId, Consu
         try {
             Optional<CachedTopic> topic = topicsCache.getTopic(qualifiedTopicName);
             if (topic.isPresent()) {
-                consumer.accept(topic.get());
-                return;
+                if (!topicsCache.isBlacklisted(qualifiedTopicName)) {
```

Comment:
@druminski that should be possible from what I see. @adamdubiel even though it's in a way non-cohesive to do that, I think for performance reasons @druminski is right on this one, any objections on your side?